core/validator_manager.go

```diff
@@ -0,0 +1,137 @@
+package core
+
+import (
+	"bytes"
+	"errors"
+	"math/big"
+	"sync"
+
+	"github.com/0xPolygon/go-ibft/messages/proto"
+)
+
+var (
+	errVotingPowerNotCorrect = errors.New("total voting power is zero or less")
+)
+
+// ValidatorManager keeps voting power and other informations about validators
+type ValidatorManager struct {
```

Comment:
Consider abstracting this with an interface, and passing the `backend` to the `ValidatorManager`, so that the `ValidatorManager` handles its state update. `ibft` can just call `Init()` on each sequence.